docs/scenarios/jdk_8_to_17_migration.md

```diff
@@ -0,0 +1,395 @@
+
+## Prerequisites
+
+Ensure you have the following set up:
+
+- [VSCode](https://code.visualstudio.com/download)
+- [Git](https://git-scm.com/downloads)
+- [GenAI credentials](https://github.com/konveyor/kai/blob/main/docs/llm_selection.md#openai-service)
+- Java 8 & Java 17 installed
+- Maven 3.9+
+- [Java sample app](https://github.com/konveyor-ecosystem/migrationex.git)
+- [Kai VSCode IDE extension 0.0.13+](https://github.com/konveyor/editor-extensions/releases)
+
+## Application Overview
+
+The [`migration example`](<(https://github.com/konveyor-ecosystem/migrationex.git)>) repository consists of two main components:
+
+1.  Java 8 Application – The sample Java 8 application (`migration example`) is a simple **Book management system** that manages a collection of books.
+2.  Rules Folder – Contains custom migration rules used by Kai for this app.
+
+### Files to modernize
+
+- `Library.java`
+- `LibraryService.java`
+- `pom.xml`
+
+## Step 1: Setup the environment
+
+### 1.1 Clone the Sample Java 8 Application
+
+Clone the migration example repository:
+
+```sh
+
+git clone https://github.com/konveyor-ecosystem/migrationex.git
+cd migrationex/migrationex
+```
+
+Navigate to File > Open in VSCode and locate the folder we just cloned and select the sub-folder `migrationex`.
+
+### 1.2 Install the Konveyor AI VS Code Extension
+
+Follow this [installation guide](https://github.com/konveyor/kai/blob/main/docs/scenarios/demo.md#configure-konveyor) to install the latest Konveyor AI IDE plugin.
+
+_Note: At the time of writing this guide, we used version 0.0.13._
+
+### 1.3 Configure Kai in VS Code
+
+1.3.1. When you launch the extension, you will land on the Welcome Page, as shown below. If the Welcome Page does not appear, proceed to the step 1.3.2.
+
+![walkthrough-1](../images/walkthrough-1.png)
+
+1.3.2. If "Set up Konveyor" is not available in the list, click the More button for additional options.
+![walkthrough-2](../images/walkthrough-2.png)
+
+1.3.3 If the welcome page does not appear, open the command palette by pressing Command + Shift + P. This will bring up a list of commands.
+![walkthrough-3](../images/walkthrough-3.png)
+
+From the command palette, locate and select the "Set up Konveyor" option. This will guide you through the configuration process.
+![walkthrough-4](../images/walkthrough-4.png)
+
+#### Configure Kai for your project.
+
+User has an option to override binaries and custom rules, however it comes with the default packaged binaries and custom rules.
+
+1. Let's select the custom rule we have
+   ![custom rule-1](../images/jdk_8_to_17_migration/customrule-1.png)
+   ![custom rule-2](../images/jdk_8_to_17_migration/customrule-2.png)
+   Make sure to select `Yes` to use the default rulesets
+   ![custom rule-3](../images/jdk_8_to_17_migration/customrule-3.png)
+   You will see a notification that custom rules updated
+   ![custom rule-4](../images/jdk_8_to_17_migration/customrule-4.png)
```

Comment:
We may want to include some guidance of what behavior is expected if a new user is running this when they do not have the RPC server running.  Below is what I see when I first bring this up on a new project for first time.  

Note the "RPC Connection is not established", assume this is normal as I don't have the RPC running yet, perhaps this is a bug.  I imagine this will be a common case where RPC server is not yet running and someone is configuring it.  The Red X implies something bad is happening.

<img width="482" alt="Screenshot 2025-03-03 at 2 08 03 PM" src="https://github.com/user-attachments/assets/6812a2c4-99c2-4b01-8b6c-fecc126e88f0" />

Reply:
I saw this too. I had some problems due to a bug that's fixed in 0.0.13. I had to go get the kai analyzer/rpc download to use newer versions of those. I wasn't sure if this was related or not in the end.

Reply:
added a section for this. Thanks for catching this issue.

Reply:
tested with 0.1.0 pre release and dont see the error anymore. removed the troubleshooting steps.